GameServerLib/Maps/SummonersRift.cs

```diff
@@ -635,15 +805,45 @@ public bool Spawn()
                 new Tuple<long, int>(35 * 60 * 1000, 0)
             };
 
-            var spawnToWaypoints = new Dictionary<string, Tuple<List<Vector2>, uint>>
+            var spawnToWaypoints = new Dictionary<string, Tuple<List<Vector2>, uint>>();
+            var waypoint = BlueMidWaypoints;
+            uint item2 = 0xffff8f1f;
+            foreach (var barrack in _mapData.SpawnBarracks)
             {
-                {Barracks.SPAWN_BLUE_BOT, Tuple.Create(BlueBotWaypoints, 0xff26ac0f)},
-                {Barracks.SPAWN_BLUE_MID, Tuple.Create(BlueMidWaypoints, 0xffff8f1f)},
-                {Barracks.SPAWN_BLUE_TOP, Tuple.Create(BlueTopWaypoints, 0xff6793d0)},
-                {Barracks.SPAWN_RED_BOT, Tuple.Create(RedBotWaypoints, 0xff9303e1)},
-                {Barracks.SPAWN_RED_MID, Tuple.Create(RedMidWaypoints, 0xff4a20f1)},
-                {Barracks.SPAWN_RED_TOP, Tuple.Create(RedTopWaypoints, 0xffd23c3e)}
-            };
+                if (barrack.Value.Name.Contains("Order"))
+                {
+                    if (barrack.Value.Name.Contains("__L"))
+                    {
+                        waypoint = BlueTopWaypoints;
+                        item2 = 0xff6793d0;
+                    }
+                    else if (barrack.Value.Name.Contains("__R"))
+                    {
+                        waypoint = BlueBotWaypoints;
+                        item2 = 0xff26ac0f;
+                    }
+                }
+                else if (barrack.Value.Name.Contains("__P_Chaos"))
+                {
+                    if (barrack.Value.Name.Contains("__C"))
+                    {
+                        waypoint = RedMidWaypoints;
+                        item2 = 0xff4a20f1;
```

Comment:
I was able to verify that these net ids are generated as follows:
CRC32(original barracks name - here it is "Barracks_T1_C1") | 0xFF000000